### PR TITLE
Tc/schedule refresh

### PR DESCRIPTION
**Pull request by CedThen**

### CODE REVIEW

File: server/routes/schedules.js [hunk 1]
```diff
@@ -177,7 +173,7 @@ router.post('/schedule/update', certify, async (req, res, next) => {
         );
       }
     });
-    return res.status(200).send(`Schedule successfully updated`);
+    return res.status(200).send(`Schedule updated successfully!`);
```
Comment (showduhtung) [inline server/routes/schedules.js:176]: lol

File: src/components/Schedule/ScheduleContainer.jsx [hunk 2]
```diff
@@ -53,16 +68,28 @@ const ScheduleContainer = ({ churchId }) => {
...
+          {alert && (
+            <Alert alert={alert} isOpen={!!alert} handleClose={() => setAlert(null)} />
+          )}
         </div>
       )}
     </div>
   );
 
+  function onDeleteSchedule(tab) {
+    console.log(`tabClone, tab`, tabClone, tab);
```
Comment (showduhtung) [inline src/components/Schedule/ScheduleContainer.jsx]: console.log

File: src/components/Schedule/Table.jsx [hunk 3]
```diff
@@ -97,56 +102,35 @@ const Table = ({
...
+                              </MuiCell>
+                              {cells.map((cell, columnIndex) => (
+                                <TableCell
+                                  cellIndices={{
+                                    serviceIndex: serviceIndex,
+                                    rowIndex: rowIndex,
+                                    columnIndex: columnIndex,
+                                  }}
+                                  roleId={roleId}
+                                  userId={cell.userId ?? 0}
```
Comment (showduhtung) [inline src/components/Schedule/Table.jsx]: I've learnt recently that it's better to keep conditional logic out of props as much as possible. You can try doing this in the TableCell component
```const TableCell = ({ ...props, userId = 0 }) => {```
This defaults `userId` to 0, if none is passed (but you'll have to check if `undefined` gets passed, I'm not sure if that affects this). 

But also only do this if this makes sense in other places we use TableCell (which I don't think we are).

File: src/components/Schedule/Table.jsx [hunk 4]
```diff
@@ -97,56 +102,35 @@ const Table = ({
...
+                                  taskId={cell.taskId ?? 0}
```
Comment (showduhtung) [inline src/components/Schedule/Table.jsx]: Same as above

File: src/components/Schedule/TableCell.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { renderOption } from './utilities';
+import { TimeCell, DutyAutocomplete, TasksAutocomplete } from '.';
+
+const TableCell = ({
+  cellIndices,
+  roleId,
+  userId,
```
Comment (showduhtung) [inline src/components/Schedule/TableCell.jsx]: so here you can make userId:
```userId = 0```

File: src/components/Schedule/useTasksAutocompleteHooks.js [hunk 6]
```diff
@@ -45,7 +43,23 @@ const useTasksAutocompleteHooks = (dataId, roleId, dataSet) => {
...
-  return [isCellModified, isCellWarning, managedDataSet, initialData];
+  useEffect(() => {
+    if (!isScheduleModified) {
+      // setIsCellWarning(false) this might be needed, have not tested it yet though
+      setIsCellModified(false);
+      setInitialData({ dataId, roleId, dataSet });
+    }
+  }, [isScheduleModified]);
+
+  const tableCellClass = isScheduleModified
```
Comment (showduhtung) [inline src/components/Schedule/useTasksAutocompleteHooks.js]: ew nested ternaries. Try to use object literals?
https://betterprogramming.pub/dont-use-if-else-and-switch-in-javascript-use-object-literals-c54578566ba0

Even though that article says not to use if-else/switch, I think even if-else or switch is better here than nested ternaries.

**Review by showduhtung — APPROVED** [review]
Take a look at the comments I mentioned. Try them out, and if you're not successful, it's fine. I think this PR is fine either way. j

File: server/routes/schedules.js (hunk 1 above)
Comment (CedThen) [reply server/routes/schedules.js:176]: lol

File: src/components/Schedule/Table.jsx (hunk 3 above)
Comment (CedThen) [reply src/components/Schedule/Table.jsx]: aw wanted to use nullish coalescing thingy but ok

**Pull request merged**